### PR TITLE
Use edfio for reading and writing edf files

### PR DESCRIPTION
**Pull request by hofaflo**
[edfio](https://github.com/the-siesta-group/edfio) could replace PyEDFlib and mne for our use cases. We would have to drop support for Python 3.8, but that is in line with the schedule we decided on in #16 anyway.

### CODE REVIEW

File: docs/examples.md [hunk 1]
```diff
@@ -86,27 +86,27 @@ plt.show()
...
-raw.crop((start - raw.info["meas_date"]).seconds, (stop - raw.info["meas_date"]).seconds)
-
-# get ECG time series as 1D NumPy array
-ecg = raw.get_data().squeeze()
+start = datetime(2023, 3, 1, 23, 0, 0)
+stop = datetime(2023, 3, 2, 6, 0, 0)
+rec_start = datetime.combine(edf.startdate, edf.starttime)
+edf.slice_between_seconds((start - rec_start).seconds, (stop - rec_start).seconds)
+
+# get ECG time series and sampling freqeuncy
```
Comment (cbrnr) [inline docs/examples.md:107]:
```suggestion
# get ECG time series and sampling frequency
```

**Review by cbrnr — COMMENTED** [review]
This is great! Thank you for this PR, I've only find a typo!

**Comment by cbrnr** [comment]
Really? We need `pooch` just to get the SciPy ECG toy data?

**Comment by hofaflo** [comment]
Apparently yes :grimacing: Do you prefer to bundle an ecg signal with the package instead?

**Comment by cbrnr** [comment]
> Do you prefer to bundle an ecg signal with the package instead?

Yes. I'd store it as compressed `.npz`, then it takes up only 163kB. We could store it in `sleepecg/data` (in analogy to `sleepecg/classifiers`)?

**Comment by cbrnr** [comment]
Maybe we could even add a little function to access the toy data (`sleepecg.get_toy_ecg()` or something similar)?

**Comment by cbrnr** [comment]
Regarding dropping support for Python 3.8, I don't think we can already remove `from __future__ import annotations`, right? This is still required for Python 3.9, or is it?

**Comment by cbrnr** [comment]
Of course the other option is to export the toy data to EDF and add `edfio` to our base requirements.

**Comment by hofaflo** [comment]
I opted for the .npz file because the .edf would be a bit larger and introduce slight signal deviations.

Ad future imports: A few have become obsolete, but most of them have to stay until 3.10, yes.

**Comment by cbrnr** [comment]
Great! I assume nothing has to be done to make sure the data gets added to the wheels, correct? We're still using `MANIFEST.in` unfortunately.

**Comment by hofaflo** [comment]
I think `graft sleepecg` makes sure this is included, yes. As cibuildwheel runs pytest after installing the built wheel, the checks would fail otherwise anyway.

**Comment by cbrnr** [comment]
Thanks @hofaflo!

**Pull request merged**